mypy/checkexpr.py

```diff
@@ -4491,8 +4491,26 @@ def visit_reveal_expr(self, expr: RevealExpr) -> Type:
             revealed_type = self.accept(
                 expr.expr, type_context=self.type_context[-1], allow_none_return=True
             )
+
+            defined = None
+            if isinstance(expr.expr, NameExpr):
+                if isinstance(expr.expr.node, Var):
+                    defined = expr.expr.node.type
+            # TODO: attributes, methods, properties
+            # elif isinstance(expr.expr, MemberExpr):
+            #     if isinstance(expr.expr.expr, NameExpr):
+            #         if isinstance(expr.expr.expr.node, Var):
+            #             ert = expr.expr.expr.node.type
+            #             if isinstance(ert, Instance):
+            #                 a = ert.type.names.get(expr.expr.name)
+            #                 if a:
+            #                     defined = a.type
```

Comment:
is there an issue for this? whats an example of it (maybe add an xfail test?)

Reply:
Ah yes, #556